go/src/koding/klient/machine/client/cached.go

```diff
@@ -0,0 +1,212 @@
+package client
+
+import (
+	"context"
+	"reflect"
+	"sync"
+	"time"
+
+	"koding/klient/machine/index"
+)
+
+// Cached allows user to cache Client method calls results. It is limited to
+// unique argument calls so it is only useful for multiple identical method
+// invocations. This type is thread safe but will not cache its results until
+// first ones are available.
+type Cached struct {
+	cu  func() (string, error)                               // CurrentUser.
```

Comment:
well since we are not encoding or decoding this vars where str len might be an issue, we'd better to use more descriptive names here